cmd/nvidia-cdi-hook/compat-libs/compat-libs.go

```diff
@@ -0,0 +1,208 @@
+/**
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package compatlibs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/NVIDIA/nvidia-container-toolkit/internal/logger"
+	"github.com/NVIDIA/nvidia-container-toolkit/internal/oci"
+)
+
+type command struct {
+	logger logger.Interface
+}
+
+type options struct {
+	folders       cli.StringSlice
+	driverVersion string
+	containerSpec string
+}
+
+// NewCommand constructs an compat-libs command with the specified logger
+func NewCommand(logger logger.Interface) *cli.Command {
+	c := command{
+		logger: logger,
+	}
+	return c.build()
+}
+
+// build the compat-libs command
+func (m command) build() *cli.Command {
+	cfg := options{}
+
+	// Create the 'compat-libs' command
+	c := cli.Command{
+		Name: "compat-libs",
```

Comment:
Should this hook have a better name?

Currently it reads:
```
nvidia-cdi-hook compat-libs --driver-version 999.88.77
```

Reply:
Changed this to:
```
nvidia-cdi-hook cuda-compat --drvier-version 999.88.77
```
We could use this hook for other compat functions such as enhanced compatibility.